pkg/config/store/fsstore2.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+)
+
+const defaultDuration = time.Second / 2
+
+type resourceMeta struct {
+	Name      string
+	Namespace string
+}
+
+type resource struct {
```

Comment:
please note that this is identical to crd/resource.

Reply:
done